src/mutation.h

```diff
@@ -137,8 +137,13 @@ struct mutation_branch {
         float hearing_modifier = 1.0f;
         float noise_modifier = 1.0f;
         float scent_modifier = 1.0f;
+        cata::optional<int> scent_intensity = cata::nullopt;
+        cata::optional<int> scent_mask = cata::nullopt;
         int bleed_resist = 0;
 
+        /**What do you smell like*/
+        scenttype_id scent_typeid;
```

Comment:
```suggestion
        cata::optional<scenttype_id> scent_typeid;
```
Than you can check it as if it was a boolean `if( something.scent_typeid )` and you don't need the empty string as sentinel.

And everybody will know that this member is sometimes simply not set to a value. Currently a reader may assume this will always contain a valid scent id.